service/src/main/kotlin/io/provenance/onboarding/domain/usecase/cee/execute/ExecuteContract.kt

```diff
@@ -0,0 +1,95 @@
+import io.provenance.onboarding.domain.usecase.provenance.account.GetAccount
+import io.provenance.onboarding.frameworks.provenance.SingleTx
+import io.provenance.onboarding.frameworks.provenance.utility.ProvenanceUtils
+import io.provenance.scope.contract.annotations.Input
+import io.provenance.scope.contract.spec.P8eContract
+import io.provenance.scope.sdk.FragmentResult
+import io.provenance.scope.sdk.SignedResult
+import mu.KotlinLogging
+import org.springframework.stereotype.Component
+import java.util.Base64
+import kotlin.reflect.full.functions
+
+private val log = KotlinLogging.logger { }
+
+@Component
+class ExecuteContract(
+    private val contractService: ContractService,
+    private val provenanceService: Provenance,
+    private val getAccount: GetAccount,
+    private val contractParser: ContractParser,
+    private val createClient: CreateClient,
+    private val getOriginator: GetOriginator,
+) : AbstractUseCase<ExecuteContractRequest, Any>() {
+
+    override suspend fun execute(args: ExecuteContractRequest): ContractExecutionResponse {
+        val utils = ProvenanceUtils()
+        val account = getAccount.execute(args.config.account)
+        val signer = utils.getSigner(account)
+        val client = createClient.execute(CreateClientRequest(args.config.account, args.config.client, args.participants))
+        val contract = contractService.getContract(args.config.contract.contractName)
+        val records = getRecords(args.records, contract)
+
+        val participants = args.participants.associate {
+            it.partyType to getOriginator.execute(it.originatorUuid)
+        }
+
+        val scope = provenanceService.getScope(args.config.provenanceConfig, args.config.contract.scopeUuid)
+        val scopeToUse: ScopeResponse? = if (scope.scope.scope.isSet() && !scope.scope.scope.scopeId.isEmpty) scope else null
+        val session = contractService.setupContract(client, contract, records, args.config.contract.scopeUuid, args.config.contract.sessionUuid, participants, scopeToUse)
+
+        return when (val result = contractService.executeContract(client, session)) {
+            is SignedResult -> {
+                provenanceService.buildContractTx(args.config.provenanceConfig, SingleTx(result))?.let {
+                    provenanceService.executeTransaction(args.config.provenanceConfig, it, signer).let { pbResponse ->
+                        ContractExecutionResponse(false, null, TxResponse(pbResponse.txhash, pbResponse.gasWanted.toString(), pbResponse.gasUsed.toString(), pbResponse.height.toString()))
+                    }
+                } ?: throw IllegalStateException("Failed to build contract for execution output.")
+            }
+            is FragmentResult -> {
+                client.requestAffiliateExecution(result.envelopeState)
+                ContractExecutionResponse(true, Base64.getEncoder().encodeToString(result.envelopeState.toByteArray()), null)
```

Comment:
Something which I'm sure you're aware of and is not necessarily relevant to this PR, but that I would like to bring up now, as these use cases get increasingly complex: in terms of the code being run, there are a ton of sources of nested exceptions that I am fairly confident are not going to get captured by (in this case) `?: throw IllegalStateException`. Running the project locally, the error codes I have seen so far have been 400 for an invalid request body and 500 for everything else, and getting more of those 500s wrapped by caller usecases into the appropriate 400 type would be good.

Reply:
Ya thats totally fair. I'll be better about handling some of these and update them as I go. Feel free to ping me with specific paths that are exposing unhandled 500s.